### ISSUE TITLE
Leading zeros with FORMAT_GEO_OUT=ddd:mm.ss.xx

### ISSUE THREAD

**Issue by anbj** [GenericMappingTools/gmt#7475]
* Based on [forumpost](https://forum.generic-mapping-tools.org/t/leading-zeros-with-format-geo-out-ddd-mm-ss-xx/3963/1).
* Will implement parts of #5153.
* There is an advantage of having fixed widths for coordinates to reduce ambiguity.
* Request: implement an option (or make if default - minimizing ambiguity pr. default is always wise) to have fixed width - i.e coordinates get leading/trailing zeros. The output in the current version is:

  ```
  $ echo 7.54353543 9.1342 | gmt convert -fg --FORMAT_GEO_OUT=ddd:mm.xxxF
  07:32.612E      09:08.052N
  ```

  Proposed new output: ``007:32.612E      09:08.052N``.

**Comment by remkos** [comment]
Were are now implementing this. In fact you pointed out a bug here, because the longitude of course should always have had 3 digits for degrees.

But there is something inconsistent between `GMT_FORMAT_OUT` and `GMT_FORMAT_MAP`. In the former `ddd` always got you leading zeros (although incorrectly as pointed out above), but in the latter it cut off the leading zeros.

For `GMT_FORMAT_MAP` we are now implementing the syntax `DDD` to also get the leading zeros, with `ddd` not giving leading zeros.

So I am suggesting to implement that *both* `GMT_FORMAT_OUT` and `GMT_FORMAT_MAP` will have two options for doing the degrees.
- `ddd` to give degrees without leading zeros (this would change the outcome for `GMT_FORMAT_OUT`).
- `DDD` to plot degrees with leading zeros. This would do the same for `GMT_FORMAT_OUT` as `ddd` did before.

@anbj : would it be a problem for you if we change this syntax for `GMT_FORMAT_OUT`?

**Comment by anbj** [comment]
@remkos: absolutely no problem. This is great.

**Comment by remkos** [comment]
Thanks. We'll make it that way then, as in PR #7655

**Comment by anbj** [comment]
Great. I added a comment in one of the commits; it says two digits for longitude and three for latitude. Unless I’m very confused I think it should be three digits for longitude (0/360 or -180/180) and two digits for latitude (-90/90).

**Comment by anbj** [comment]
Fixed in 9e8e200.

**Comment by remkos** [comment]
> Fixed in [9e8e200](https://github.com/GenericMappingTools/gmt/commit/9e8e200d7d5ae048d4d53447f36b57db39d5f722).

Thanks. Yes, I got confused somewhere.
